[Live Chat] Update and fine-tune message reaction functionality and UI

pmdepaulo opened the pull request:
Commented out buggy test code for now but will make sure to add it back so that merges go smoothly!

aflekkas committed:
Added profanity toggle and url for profanity toggle (now we need to do it visually too)

aflekkas committed:
Added a button to toggle the profanity filter in the user profile detail view.

pmdepaulo committed:
[live-chat] WIP: reinforcing reaction functionality + added better button UI

pmdepaulo committed:
[live-chat] WIP: updating views to accomodate live chat reactions

pmdepaulo committed:
[live-chat] made a slightly buggy version of reacting to messages (and having them stay)

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

pmdepaulo committed:
[live-chat] fixed buggy message reactions... kind of

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

pmdepaulo commented:
Added the functionality of maintaining chat reactions on reload by loading them in with the message rendering. Could be improved, however, by making sure that they're clickable on reload and not just displayed.

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

pmdepaulo committed:
Merge remote-tracking branch 'refs/remotes/origin/live-chat/update-reaction-UI' into live-chat/update-reaction-UI

pmdepaulo committed:
Merge remote-tracking branch 'refs/remotes/origin/live-chat/update-reaction-UI' into live-chat/update-reaction-UI

pmdepaulo commented:
Slightly buggy still because it allows you to click on the reaction pill (and update it) for reactions that you've made in that same session, but not for ones that still exist in the db.

nathanL534 reviewed (changes requested):
This looks really good, and I tested everything locally and almost all the functions work really smoothly. There is a bug with edit messages, but thats already fixed on another branch, #1039.  It. might be a make sense to merge into #1039, as that has 0 merge conflicts and will help us avoid redundant fixes across PRs. Im also happy to hop on later and help get this sorted.

quincyleung committed:
Improve toggle profanity UI

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

umerp committed:
[Tournaments] Add Live Character Counters and Character Limit to Tournament Creation (#1373)

* Add knowledge base migration before merging dev

* I have added  Live Character Counters and Character Limit to make sure games can be started with ease resloves issue #1366

* Remove unnecessary migration file from knowledge_base

* Trigger GitHub conflict resolution refresh

* fixed issues

Abebes01 committed:
[games] Implement weighted recommendation algorithm (#1121)

* [games] Implement weighted recommendation algorithm

Added new function that recommends similar games based on
multiple weighted factors including categories, mechanics,
and designers.

Closes: #874

* trigger comment

---------

Co-authored-by: ekabaeva <122409534+ekabaeva@users.noreply.github.com>
Co-authored-by: Sam Xie <72187977+MuyanXie@users.noreply.github.com>

quincyleung committed:
Merged dev into branch and resolved conflict. Ready to re-add toggle UI

quincyleung committed:
Fix profanity toggle logic and conflicts

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

pmdepaulo committed:
[live-chat] linter fix pt 2 - editing template

pmdepaulo committed:
[live-chat] fixing more linter issues

pmdepaulo committed:
[live-chat] fixed bug with reaction functionality

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

quincyleung committed:
Fix merge conflicts

quincyleung committed:
Fix linting errors

quincyleung committed:
Add docstrings for profanity toggling

Commit:
[live chat] fixed merged issues and broken features

nathanL534 reviewed (approved):
Tested it locally, everything works very well, and all the previous components' functionality are working. Great work. It seems like everything is up to date with the development branch as well. Awesome job, let's ship it!

pmdepaulo committed:
Merge branch 'dev' into live-chat/update-reaction-UI

abargher reviewed (approved):
Looks great!

abargher commented:
Issue Score: Excellent

Comments:
Great work, good status updates!

<!-- {"score": "excellent", "minor": false} -->
